### ISSUE TITLE
Any plans to switch to GTK3 Header Bars?

### ISSUE THREAD

**Issue by jurf** [lwindolf/liferea#176]
It would save a lot of screen space and give a Liferea a fresh new look.

Anyway, just an idea.

**Comment by lwindolf** [comment]
We will probably migrate some day. Just found no time to convert the code. I estimate it needs a lot of reorganization.

**Comment by DoctorJellyface** [comment]
Great, thanks!

**Comment by gusnan** [comment]
PLEASE make this optional. For me (and I belive many others who doesn't use Gnome3), the new GTK3 stuff is simply a no-go.

**Comment by asl97** [comment]
@gusnan it already recommenced for liferea to be build against gtk3, in fact, gtk3 is listed under mandatory.

https://github.com/lwindolf/liferea#dependencies
https://github.com/lwindolf/liferea#mandatory

also, iirc, there are lesser and lesser distro being ship with Gnome2 and
GTK3 isn't that new anymore (it was release in 2011).

**Comment by gusnan** [comment]
Does that mean it _requires_ the new style of titlebars?

**Comment by DoctorJellyface** [comment]
No, but AFAIK client side decorations have support with almost every window manager out there. Wayland even requires CSD.

**Comment by lwindolf** [comment]
@gusnan In the scope of Liferea (one man project) supporting two variants isn't possible.

**Comment by gusnan** [comment]
Well, then the logical thing would be to support the alternative that requires the least changes, and not a redesign. But, if a majority want the new stuff, then I'll simply will have to accept it.

**Comment by lwindolf** [comment]
@gusnan Don't worry GNOME 4 will have even more "new" stuff for sure! ;-)

**Comment by gnumdk** [comment]
On demand headerbar is easy, here what i've done in lollypop (python code)
```python
        toolbar = Gtk.HeaderBar()
        # Only set headerbar if according DE detected or forced manually
        if use_csd():
            self.set_titlebar(toolbar)
            self._toolbar.set_show_close_button(True)
            self.add(main_widget())
        else:
            hgrid = Gtk.Grid()
            hgrid.set_orientation(Gtk.Orientation.VERTICAL)
            hgrid.add(toolbar)
            hgrid.add(main_widget)
            hgrid.show()
            self.add(hgrid)
```
Most of the work is about redesign app to get ride of menus ;)

**Comment by tpo** [comment]
FWIW - I also wouldn't welcome the GTK3 header bars, because:

1) at least my window manager (XFce4) - why I don't know - is unable to handle them properly handle GTK3 header bars, that is:

1.a) I can't move the new style GTK3 header bar applications to the top of the screen, XFce apparently thinks they are larger than they really are and so I can't move them out of "the middle of the screen" despite me not wanting to having them there in my face

1.b) when I move them to the border the applications start resizing to full screen, which is usually completely unexpected, annoying and unasked for and not their job (IMO) but the job of the window manager (and its configurable behavior)

2) the header bars are *fat* and use a lot of screen estate for some aesthetical reason that I personally do not care about, since I want to have as much functional screen estate as possible.

I guess all these things are configurable somehow, but actually my goal wrt to applications is to be using them and not to be spending time tweaking their UI, and increasingly it seems like every toolkit feels the need to have some own special sauce for its configuration so I need tools and special settings for old GTK applications, for new GTK3 applications, for KDE applications and in the end everything is still behaving and looking inconsistently, which in my eyes is not an improvement.

So I'd be glad if liferea wouldn't switch.

That being said, there is still some faint hope, as it seems it's maybe possible to switch CSD completely off: https://bugs.debian.org/788076

Anyway, power to the doers, please don't let me prevent you from moving in the direction you think is right and thanks a lot for liferea!

**Comment by asl97** [comment]
1a: use alt-click-drag to move
1b: are you sure it isn't the Window tiling mode?
1b: it does that when it is maximized when you move it
1b: it could also be just the app and not gtk3
2: just use alt-click-drag and move it off screen

**Comment by DoctorJellyface** [comment]
AFAIK Xfce 4.12 [has perfect](http://blog.alteroot.org/articles/2015-02-19/new-from-xfce-part-3.html) for CSD, window manager, theming and all.

Also, header bars are a replacement for the title, menu and tool bars, so in the end it's IMHO much thinner.

**Comment by Leiaz** [comment]
I've looked at moving the GtkActions to GAction and the menu to an application menubar and modifying the associated deprecated stuff. I think that could be a first step for this.

It is not finished, and some things may be broken. I don't know Gtk very well. I was mostly curious, and I didn't know if I would actually get anywhere ... I pushed it on a gaction branch on my fork. @lwindolf  if you have a different plan for that refactoring, I could try to help.

Also, I use XMonad with pieces of Xfce, and I don't like the "tablet" look that is contaminating everything. I like real menus, not the big squares for fat fingers that some applications have ... so from what I have seen so far, I'm not a big fan of that header :)

**Comment by lwindolf** [comment]
1.12.3 will include an optional plugin that loads Liferea with a GtkHeaderBar dynamically. Implementation in #614, discussion (and screenshot) in #276.

**Comment by lwindolf** [comment]
Headerbar plugin was merged for 1.12.3. Closing this issue as solved. As state in #614 the plugin will stay optional and opt-in only during 1.12 so we can collect feedback on it.

**Comment by Gert-dev** [comment]
Thanks for finally merging it, looks great!
